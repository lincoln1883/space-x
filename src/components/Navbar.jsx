import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/planet.png';

const Navbar = () => (
  <>
    <nav className="navbar navbar-expand-lg">
      <div className="container-fluid">
        <NavLink className="navbar-brand ms-4 d-flex align-items-center" to="/">
          <img src={logo} alt="planet" width="70" height="70" />
          <h1 className="ms-2 fs-3 d-none d-sm-block">
            Space Travelers&apos; Hub
          </h1>
        </NavLink>
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarScroll" aria-controls="navbarScroll" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarScroll">
          <ul className="navbar-nav ms-auto my-2 my-lg-0 navbar-nav-scroll">
            <li className="nav-item">
              <NavLink className="nav-link text-primary" aria-current="page" to="/">Rockets</NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link  text-primary" to="/missions">Missions</NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link  text-primary" to="/profile">Profile</NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <hr className="w-auto ms-5 me-5" />
  </>
);

export default Navbar;
