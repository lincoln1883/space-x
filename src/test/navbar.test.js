import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';
import '@testing-library/jest-dom/extend-expect';

describe('Navbar', () => {
  it('renders Navbar component', () => {
    const tree = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('Navbar', () => {
  it('renders Navbar component', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );

    const navbarElement = screen.getByText(/Space Travelers' Hub/i);
    expect(navbarElement).toBeInTheDocument();
    expect(navbarElement).toHaveClass('ms-2');

    const rocketsLink = screen.getByText(/Rockets/i);
    const missionsLink = screen.getByText(/Missions/i);
    const profileLink = screen.getByText(/Profile/i);

    expect(rocketsLink).toBeInTheDocument();
    expect(missionsLink).toBeInTheDocument();
    expect(profileLink).toBeInTheDocument();
  });
  it('should render the logo', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    const logo = screen.getByAltText(/planet/i);
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute('src', 'planet.png');
  });
  it('should should have links to paths', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    const rocketsLink = screen.getByText(/Rockets/i);
    const missionsLink = screen.getByText(/Missions/i);
    const profileLink = screen.getByText(/Profile/i);
    expect(rocketsLink).toHaveAttribute('href', '/');
    expect(missionsLink).toHaveAttribute('href', '/missions');
    expect(profileLink).toHaveAttribute('href', '/profile');
  });
});
