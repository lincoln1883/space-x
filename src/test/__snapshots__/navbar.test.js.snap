// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg"
      >
        <div
          class="container-fluid"
        >
          <a
            aria-current="page"
            class="navbar-brand ms-4 d-flex align-items-center active"
            href="/"
          >
            <img
              alt="planet"
              height="70"
              src="planet.png"
              width="70"
            />
            <h1
              class="ms-2 fs-3 d-none d-sm-block"
            >
              Space Travelers' Hub
            </h1>
          </a>
          <button
            aria-controls="navbarScroll"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-bs-target="#navbarScroll"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarScroll"
          >
            <ul
              class="navbar-nav ms-auto my-2 my-lg-0 navbar-nav-scroll"
            >
              <li
                class="nav-item"
              >
                <a
                  aria-current="page"
                  class="nav-link text-primary active"
                  href="/"
                >
                  Rockets
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link  text-primary"
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link  text-primary"
                  href="/profile"
                >
                  Profile
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <hr
        class="w-auto ms-5 me-5"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand ms-4 d-flex align-items-center active"
          href="/"
        >
          <img
            alt="planet"
            height="70"
            src="planet.png"
            width="70"
          />
          <h1
            class="ms-2 fs-3 d-none d-sm-block"
          >
            Space Travelers' Hub
          </h1>
        </a>
        <button
          aria-controls="navbarScroll"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarScroll"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarScroll"
        >
          <ul
            class="navbar-nav ms-auto my-2 my-lg-0 navbar-nav-scroll"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link text-primary active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link  text-primary"
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link  text-primary"
                href="/profile"
              >
                Profile
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <hr
      class="w-auto ms-5 me-5"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
